### PR TITLE
Fix #568: object is unordered

### LINKED ISSUE #568: aeson testsuite failures on 32bit and Big Endian architectures
Hello, seems that HashMap test is failing on i386 and other architectures:
```
    DList: [OK]
    (): [OK]
    HashMap Int Int: [Failed]
encode/via value
expected: "{\"0\":1,\"2\":3}"
 but got: "{\"2\":3,\"0\":1}"
    Map Int Int: [Failed]
encode/via value
expected: "{\"0\":1,\"2\":3}"
 but got: "{\"2\":3,\"0\":1}"
    Map (Tagged Int Int) Int: [Failed]
encode/via value
expected: "{\"0\":1,\"2\":3}"
 but got: "{\"2\":3,\"0\":1}"
    Map [Int] Int: [OK]
    Map [Char] Int: [Failed]
encode/via value
expected: "{\"ab\":1,\"cd\":3}"
 but got: "{\"cd\":3,\"ab\":1}"
    Map [I Char] Int: [Failed]
encode/via value
expected: "{\"ab\":1,\"cd\":3}"
 but got: "{\"cd\":3,\"ab\":1}"
    nan :: Double: [OK]

```

do you have any clue?
build logs available here
https://buildd.debian.org/status/package.php?p=haskell-aeson&suite=unstable

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

File: tests/SerializationFormatSpec.hs [hunk 1]
```diff
@@ -65,132 +66,144 @@ tests =
...
+  , example "NonEmpty" "[1,2,3]"  (1 :| [2, 3] :: NonEmpty Int)
+  , example "Seq" "[1,2,3]"  (Seq.fromList [1, 2, 3] ::  Seq.Seq Int)
+  , example "DList" "[1,2,3]"  (DList.fromList [1, 2, 3] :: DList.DList Int)
+  , example "()" "[]"  ()
+
+  , Example "HashMap Int Int"
+        [ "{\"0\":1,\"2\":3}", "{\"2\":3,\"0\":1}"]
+        (HM.fromList [(0,1),(2,3)] :: HM.HashMap Int Int)
+  , Example "Map Int Int"
+        [ "{\"0\":1,\"2\":3}", "{\"2\":3,\"0\":1}"]
```
Comment (phadej) [inline tests/SerializationFormatSpec.hs]: as we have examples of objects with only two keys, listing all permutations is viable

File: tests/SerializationFormatSpec.hs [hunk 2]
```diff
@@ -65,132 +66,144 @@ tests =
...
+        (M.fromList [(0,1),(2,3)] :: M.Map Int Int)
+  , Example "Map (Tagged Int Int) Int"
+        [ "{\"0\":1,\"2\":3}", "{\"2\":3,\"0\":1}"]
+        (M.fromList [(Tagged 0,1),(Tagged 2,3)] :: M.Map (Tagged Int Int) Int)
+  , example "Map [Int] Int"
+        "[[[0],1],[[2],3]]"
```
Comment (phadej) [inline tests/SerializationFormatSpec.hs]: `Map` is ordered! and list maintains that order.

**Comment by bergmark** [comment]
Thanks! But Is this supposed to only contain changes to the test suite?

**Comment by bergmark** [comment]
Oh n/m #568 was a test suite issue :-)

**Pull request merged**